src/test/scala/com/atomist/rug/runtime/js/TypeScriptRugReviewerTest.scala

```diff
@@ -0,0 +1,212 @@
+import com.atomist.rug.TestUtils
+import com.atomist.rug.compiler.typescript.TypeScriptCompiler
+import com.atomist.source.{FileArtifact, SimpleFileBasedArtifactSource, StringFileArtifact}
+import org.scalatest.{FlatSpec, Matchers}
+
+object TypeScriptRugReviewerTest {
+
+  val ContentPattern = "^Anders .*$"
+
+  val compiler = new TypeScriptCompiler()
+
+  val ParameterContent = "Anders Hjelsberg is God"
+
+  val SimpleReviewerWithoutParameters =
+    """
+      |import {Project} from '@atomist/rug/model/Core'
+      |import {ProjectReviewer} from '@atomist/rug/operations/ProjectReviewer'
+      |import {ReviewResult,ReviewComment} from '@atomist/rug/operations/RugOperation'
+      |
+      |class SimpleReviewer implements ProjectReviewer {
+      |    name: string = "Simple"
+      |    description: string = "My simple editor"
+      |    review(project: Project):ReviewResult {
+      |        return new ReviewResult("",
+      |        <ReviewComment[]>[])
+      |    }
+      |}
+      |var reviewer = new SimpleReviewer()
+    """.stripMargin
+
+  val SimpleReviewerWithParameters =
+      s"""
+         |import {Project} from '@atomist/rug/model/Core'
+         |import {ProjectReviewer} from '@atomist/rug/operations/ProjectReviewer'
+         |import {File} from '@atomist/rug/model/Core'
+         |import {ReviewResult, ReviewComment, Parameter} from '@atomist/rug/operations/RugOperation'
+         |
+       |class SimpleReviewer implements ProjectReviewer {
+         |    name: string = "Simple"
+         |    description: string = "A nice little reviewer"
+         |    parameters: Parameter[] = [{name: "content", description: "Content", pattern: "@url", maxLength: 100}]
+         |    review(project: Project, {content} : {content: string}) {
+         |      //p["otherParam"] = p.content
+         |      return new ReviewResult(content,
+         |          <ReviewComment[]>[]
+         |        );
+         |    }
+         |  }
+         |var reviewer = new SimpleReviewer()
+    """.stripMargin
+
+  val SimpleReviewerWithParametersSinglePartialCommentReviewResult =
+    s"""
+       |import {Project} from '@atomist/rug/model/Core'
+       |import {ProjectReviewer} from '@atomist/rug/operations/ProjectReviewer'
+       |import {File} from '@atomist/rug/model/Core'
+       |import {ReviewResult, ReviewComment, Parameter, Severity} from '@atomist/rug/operations/RugOperation'
+       |
+       |class SimpleReviewer implements ProjectReviewer {
+       |    name: string = "Simple"
+       |    description: string = "A nice little reviewer"
+       |    parameters: Parameter[] = [{name: "content", description: "Content", pattern: "@url", maxLength: 100}]
+       |    review(project: Project, {content} : {content: string}) {
+       |      //p["otherParam"] = p.content
+       |      return new ReviewResult(content,
+       |          <ReviewComment[]>[new ReviewComment(content, Severity.Broken)]
```

Comment:
In this case, probably not. Only necessary I think when it was an empty array. I'll look to remove.